src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,82 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
+    private static final Map<Class<?>, Class<?>> interfaceImplementations = Map.of(
```

Comment:
Your code is clean and informative, excellent job) just a small remark: you can move this interfaceImplementations Map to findImplementation method as was described in video (cause we gonna use this map only in this method)

Reply:
I think it's more convenient to leave it in this way)

Reply:
this implementation is ok.